### PR TITLE
feat: adding revoke all claims of a connection

### PR DESCRIPTION
**Pull request by javip97**
Revoking all claims of a connection will revoke all the non revoked claims of the given connection.

The revoking claims works as a transaction, or every claim get revoked or non get revoked

### CODE REVIEW

File: internal/repositories/claims.go [hunk 1]
```diff
@@ -403,6 +403,44 @@ func (c *claims) GetAllByIssuerID(ctx context.Context, conn db.Querier, identifi
...
+			`
+
+	rows, err := conn.Query(ctx, query, connID.String(), issuerID.String())
+	if err != nil {
+		if err == pgx.ErrNoRows {
+			return []*domain.Claim{}, nil // connections can have 0 credentials
+		}
+
+		return nil, err
+	}
```
Comment (x1m3) [inline internal/repositories/claims.go:438]:
```suggestion
	if err != nil && err != pgx.ErrNoRows {
		return nil, err
	}
```

File: internal/api_admin/server.go [hunk 2]
```diff
@@ -325,3 +300,38 @@ func (s *Server) PublishState(ctx context.Context, request PublishStateRequestOb
 		TxID:               publishedState.TxID,
 	}, nil
 }
+
+// RevokeConnectionCredentials revoke all the non revoked credentials of the given connection
+func (s *Server) RevokeConnectionCredentials(ctx context.Context, request RevokeConnectionCredentialsRequestObject) (RevokeConnectionCredentialsResponseObject, error) {
+	err := s.claimService.RevokeAllFromConnection(ctx, request.Id, s.cfg.APIUI.IssuerDID)
+	if err != nil {
```
Comment (x1m3) [inline internal/api_admin/server.go:307]: log the error

File: internal/api_admin/server.go [hunk 3]
```diff
@@ -296,7 +271,7 @@ func (s *Server) CreateCredential(ctx context.Context, request CreateCredentialR
 
 // RevokeCredential - revokes a credential per a given nonce
 func (s *Server) RevokeCredential(ctx context.Context, request RevokeCredentialRequestObject) (RevokeCredentialResponseObject, error) {
-	if err := s.claimService.Revoke(ctx, s.cfg.APIUI.IssuerDID.String(), uint64(request.Nonce), ""); err != nil {
+	if err := s.claimService.Revoke(ctx, s.cfg.APIUI.IssuerDID, uint64(request.Nonce), ""); err != nil {
```
Comment (x1m3) [inline internal/api_admin/server.go:274]: log the errror

File: internal/api/server.go [hunk 4]
```diff
@@ -134,7 +134,12 @@ func (s *Server) CreateClaim(ctx context.Context, request CreateClaimRequestObje
 
 // RevokeClaim is the revocation claim controller
 func (s *Server) RevokeClaim(ctx context.Context, request RevokeClaimRequestObject) (RevokeClaimResponseObject, error) {
-	if err := s.claimService.Revoke(ctx, request.Identifier, uint64(request.Nonce), ""); err != nil {
+	did, err := core.ParseDID(request.Identifier)
+	if err != nil {
+		return RevokeClaim400JSONResponse{N400JSONResponse{err.Error()}}, nil
```
Comment (x1m3) [inline internal/api/server.go:139]: log the error

**Review by x1m3 — CHANGES_REQUESTED** [review]
LGTM.  Just some small comments regarding logs.

File: internal/repositories/claims.go (hunk 1 above)
Comment (javip97) [reply internal/repositories/claims.go:438]: agree, I will change the defer as well 

```
defer rows.Close()
	
if err != nil && err != pgx.ErrNoRows {
	return nil, err
}
```

File: internal/api_admin/server.go (hunk 3 above)
Comment (javip97) [reply internal/api_admin/server.go:274]: thanks

File: internal/api/server.go (hunk 4 above)
Comment (javip97) [reply internal/api/server.go:139]: thanks

File: internal/api_admin/server.go (hunk 2 above)
Comment (javip97) [reply internal/api_admin/server.go:307]: thanks

**Pull request merged**